### DISCUSSION TITLE
Support for static clients

### DISCUSSION THREAD

**Discussion by benfuu** [zitadel/zitadel#5909] [Q&A]
I've been using Zitadel for a while to play around with in my homelab and have been loving all the features and integrations! I'm currently trying to integrate OwnCloud OCIS. I saw that for native (Desktop, Mobile) clients, OwnCloud ships their own hardcoded [client id and secret](https://doc.owncloud.com/server/next/admin_manual/configuration/user/oidc/oidc.html#client-id).

Is it possible to configure these in Zitadel? I've tried editing an existing application and giving it a custom client ID but it doesn't allow me to save the changes so I am not sure if it is possible.

**Comment by hifabienne** [comment]
Hei @bennycooly 
This is currently not possible.
At the moment ZITADEL always generates the client id and secret for you.
If we see that this feature is something that is requested by a lot of customers we will have a look at it.
